test/acceptance/tests/basic/basic_test.go

```diff
@@ -30,134 +30,178 @@ func TestValidation_CACertRequiredIfTLSIsEnabled(t *testing.T) {
-					},
-					Logger: terratestLogger.New(logger.TestLogger{}),
-				})
-				r.Check(err)
-				if !strings.Contains(out, fmt.Sprintf("test_client_%s", randomSuffix)) ||
-					!strings.Contains(out, fmt.Sprintf("test_server_%s", randomSuffix)) {
-					r.Errorf("services not yet registered, got %q", out)
-				}
-			})
-
-			// use aws exec to curl between the apps
-			taskListOut := shell.RunCommandAndGetOutput(t, shell.Command{
-				Command: "aws",
-				Args: []string{
-					"ecs",
-					"list-tasks",
-					"--region",
-					suite.Config().Region,
-					"--cluster",
-					suite.Config().ECSClusterARN,
-					"--family",
-					fmt.Sprintf("test_client_%s", randomSuffix),
-				},
-			})
-
-			var tasks listTasksResponse
-			require.NoError(t, json.Unmarshal([]byte(taskListOut), &tasks))
-			require.Len(t, tasks.TaskARNs, 1)
-
-			retry.RunWith(&retry.Timer{Timeout: 3 * time.Minute, Wait: 10 * time.Second}, t, func(r *retry.R) {
-				curlOut, err := shell.RunCommandAndGetOutputE(t, shell.Command{
-					Command: "aws",
-					Args: []string{
-						"ecs",
-						"execute-command",
-						"--region",
-						suite.Config().Region,
-						"--cluster",
-						suite.Config().ECSClusterARN,
-						"--task",
-						tasks.TaskARNs[0],
-						"--container=basic",
-						"--command",
-						`/bin/sh -c "curl localhost:1234"`,
-						"--interactive",
-					},
-					Logger: terratestLogger.New(logger.TestLogger{}),
-				})
-				r.Check(err)
-				if !strings.Contains(curlOut, `"code": 200`) {
-					r.Errorf("response was unexpected: %q", curlOut)
-				}
-			})
-
-			logger.Log(t, "Test successful!")
+			doBasicInstall(t, secure, "EC2")
 		})
 	}
 }
 
+// Helper to deploy the `basic-install` Terraform module, which will
+// deploy a sample client-server application in an ECS cluster.
+func doBasicInstall(t *testing.T, secure bool, launchType string) {
+	// Use separate clusters for FARGATE / EC2 to run in parallel.
+	var clusterArn string
+	if launchType == "FARGATE" {
+		clusterArn = suite.Config().ECSClusterARN_Fargate
+	} else if launchType == "EC2" {
+		clusterArn = suite.Config().ECSClusterARN_EC2
+	} else {
+		t.Fatalf("invalid launch type: %s", launchType)
+	}
+	if clusterArn == "" {
+		t.Fatalf("no cluster ARN for %s launch type", launchType)
+	}
+
+	randomSuffix := strings.ToLower(random.UniqueId())
+	tfVars := suite.Config().TFVars()
+	tfVars["secure"] = secure
+	tfVars["suffix"] = randomSuffix
+	tfVars["launch_type"] = launchType
+	tfVars["ecs_cluster_arn"] = clusterArn
+
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: "./terraform/basic-install",
+		Vars:         tfVars,
+		NoColor:      true,
+		EnvVars:      map[string]string{},
+	})
+	terraform.Init(t, terraformOptions)
+
+	// Unique-ify state files by launch type since we parallelize by launch type.
+	// This means only two state files to deal with in case of manual cleanup locally.
+	stateFile := fmt.Sprintf("terraform-%s.tfstate", launchType)
```

Comment:
Calling this out: separate Terraform state files in order to run terraform in parallel.

terratest has a [CopyTerraformFolderToTemp](https://pkg.go.dev/github.com/gruntwork-io/terratest/modules/test-structure#CopyTerraformFolderToTemp) function, that can copy the `basic-install` module to a temp dir. But, it doesn't fix relative module paths, and makes manual cleanup difficult ("where's that temp dir with the Terraform state file?")